File: doc-examples/example-java/src/test/java/example/SandboxTest.java
package example;

import io.micronaut.el.CompiledELContext;
import io.micronaut.el.ELSandbox;
import io.micronaut.el.ELSandboxException;
import jakarta.el.ELContext;
import jakarta.el.ELManager;
import jakarta.el.ValueExpression;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class SandboxTest {

    @Test
    void theSandboxCanBeWidened() {
        ELContext context = new CompiledELContext().setBean("book", new Book("Jakarta EL", "reference", 20d));
        ValueExpression expression = ELManager.getExpressionFactory()
            .createValueExpression(context, "${book.class.name}", String.class); // <1>
        assertThrows(ELSandboxException.class, () -> expression.getValue(context)); // <2>

        context.putContext(ELSandbox.class, ELSandbox.UNRESTRICTED); // <3>
        assertEquals("example.Book", expression.getValue(context));
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/SandboxTest.kt
package example

import io.micronaut.el.CompiledELContext
import io.micronaut.el.ELSandbox
import io.micronaut.el.ELSandboxException
import jakarta.el.ELManager
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class SandboxTest {

    @Test
    fun theSandboxCanBeWidened() {
        val context = CompiledELContext().setBean("book", Book("Jakarta EL", "reference", 20.0))
        val expression = ELManager.getExpressionFactory()
            .createValueExpression(context, "\${book.class.name}", String::class.java) // <1>
        assertThrows(ELSandboxException::class.java) { expression.getValue(context) } // <2>

        context.putContext(ELSandbox::class.java, ELSandbox.UNRESTRICTED) // <3>
        assertEquals("example.Book", expression.getValue(context))
    }
}
